x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PropagateEvalFoldables.java

```diff
@@ -48,10 +54,20 @@ public LogicalPlan apply(LogicalPlan plan) {
 
         plan = plan.transformUp(p -> {
             // Apply the replacement inside Filter and Eval (which shouldn't make a difference)
-            // TODO: also allow aggregates once aggs on constants are supported.
-            // C.f. https://github.com/elastic/elasticsearch/issues/100634
             if (p instanceof Filter || p instanceof Eval) {
                 p = p.transformExpressionsOnly(ReferenceAttribute.class, replaceReference);
+            } else if (p instanceof Aggregate agg) {
+                List<NamedExpression> newAggs = new ArrayList<>(agg.aggregates().size());
+                agg.aggregates().forEach(e -> {
+                    if (Alias.unwrap(e) instanceof AggregateFunction af && af.isConstantFoldable()) {
```

Comment:
`isConstantFoldable()` to skip geo aggregate functions. Otherwise, AbstractPhysicalOperationProviders will complain that  the geo functions having a foldable value as arguments reached [a point they shouldn't have](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AbstractPhysicalOperationProviders.java#L248).